### ISSUE TITLE
Broken link in the exception message

### ISSUE THREAD

**Issue by Wajihulhassan** [eclipse-che/che#1004]
In the thrown exception message at [FactoryService.java#L161](https://github.com/eclipse/che/blob/85b69e0ebfe5b5bb30f66eba12eb5b0c96796930/wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java#L161), the link http://docs.codenvy.com/user/creating-factories/factory-parameter-reference/ is broken.

**Comment by TylerJewell** [comment]
Wow @Wajihulhassan what a find. I wasn't even aware that the Factory Service was embedded within Che yet. It was going to be a bit of a surprise to the community when we move the Factory stuff from within Codenvy into Che.

@skabashnyuk - in the upcoming sprint, please have a short issue to have us update all of the error messages. I went through all of the error messages in the file and updated them to be more readable. I also pulled out any reference to documentation pages, which isn't a great practice.  Some of the error codes are not consistently applied for all methods, so let's go ahead and make them consistent.

```
code = 200, message = "OK"
code = 400, message = "Parameters are not valid: missing required parameters"
code = 403, message = "You do not have the permissions to to perform a Factory save operation"
code = 409, message = "Parameters are not valid: missing parameters causing conflict"
updatefactory: code = 409, message = "The factory information is not updateable"
code = 500, message = "Unable to identify user from context"
row 161, BadRequestException, message = "You have provided an invalid JSON"
```

**Comment by TylerJewell** [comment]
@Wajihulhassan - what are you working on at university?  I got my CS degree 20 years ago at Rose-Hulman in Terre Haute, IN.

**Comment by Wajihulhassan** [comment]
@TylerJewell  As a part of my course project I am testing Che for bugs and reporting them to developers.

**Comment by TylerJewell** [comment]
Please let us know if there are ways that we can improve the CONTRIBUTING.md content and approach. What you are doing is a tremendous help.

**Comment by TylerJewell** [comment]
@skabashnyuk - I believe we resolved all of this in 4.2 or 4.3.  Please confirm.

**Comment by mshaposhnik** [comment]
#1702
